src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -134,6 +180,17 @@ private void initialize(String sasKeyName, String sasKey, String host, ServiceBu
 
             return properties;
         });
+        
+        if (isToken) {
+        	this.factory = new JmsConnectionFactory(destinationUri);
+        	this.factory.setExtension(JmsConnectionExtensions.USERNAME_OVERRIDE.toString(), (connection, uri) -> {        		
+        		return "$jwt";
+        	});
+            this.factory.setExtension(JmsConnectionExtensions.PASSWORD_OVERRIDE.toString(), GetToken);
```

Comment:
>GetToken [](http://example.com/codeflow?start=92&length=8)

nit: this.GetToken()

Reply:
separate this into setAadExtentions since it only applies to aad and us isAadAuthentication to make distinction btw sas and aad